app-foundation/3-networks/development/firewall.tf

```diff
@@ -0,0 +1,311 @@
+ VPC firewall rules
+*****************************************/
+
+locals {
+  boa_gke_cluster1_master_cidr = "100.64.78.0/28"
+  boa_gke_cluster2_master_cidr = "100.65.70.0/28"
+  boa_gke_mci_master_cidr      = "100.64.70.0/28"
+}
+
+resource "google_compute_firewall" "allow_lb_healthcheck" {
+  name      = "fw-${var.environment_code}-shared-base-allow-lb-healthcheck"
+  project   = local.base_project_id
+  network   = module.base_shared_vpc.network_self_link
+  priority  = 1000
+  direction = "INGRESS"
+
+  dynamic "log_config" {
+    for_each = var.firewall_enable_logging == true ? [{
+      metadata = "INCLUDE_ALL_METADATA"
+    }] : []
+
+    content {
+      metadata = log_config.value.metadata
+    }
+  }
+
+  source_ranges           = ["35.191.0.0/16", "130.211.0.0/22"]
+  target_service_accounts = ["tf-gke-gke-boa-us-east-huj0@prj-bu1-d-boa-gke-ecb0.iam.gserviceaccount.com"]
+
+  allow {
+    protocol = "tcp"
+    ports    = ["443", "10250", "15017"]
+  }
+}
+
+resource "google_compute_firewall" "allow_asm_healthcheck_sidecar_east" {
+  name      = "fw-${var.environment_code}-shared-base-allow-asm-healthcheck-autosidecar-east"
+  project   = local.base_project_id
+  network   = module.base_shared_vpc.network_self_link
+  priority  = 1000
+  direction = "INGRESS"
+
+  dynamic "log_config" {
+    for_each = var.firewall_enable_logging == true ? [{
+      metadata = "INCLUDE_ALL_METADATA"
+    }] : []
+
+    content {
+      metadata = log_config.value.metadata
+    }
+  }
+
+  source_ranges = ["172.16.2.0/28"]
+  target_tags   = ["gke-gke-boa-us-east1-001-1d5c9eff-node"]
+
+  allow {
+    protocol = "tcp"
+    ports    = ["443", "10250", "15017"]
+  }
+}
+
+resource "google_compute_firewall" "allow_asm_healthcheck_sidecar_west" {
+  name      = "fw-${var.environment_code}-shared-base-allow-asm-healthcheck-autosidecar-west"
+  project   = local.base_project_id
+  network   = module.base_shared_vpc.network_self_link
+  priority  = 1000
+  direction = "INGRESS"
+
+  dynamic "log_config" {
+    for_each = var.firewall_enable_logging == true ? [{
+      metadata = "INCLUDE_ALL_METADATA"
+    }] : []
+
+    content {
+      metadata = log_config.value.metadata
+    }
+  }
+
+  source_ranges = ["172.16.0.16/28"]
+  target_tags   = ["gke-gke-boa-us-west1-001-20e58b91-node"]
+
+  allow {
+    protocol = "tcp"
+    ports    = ["443", "10250", "15017"]
+  }
+}
+
+resource "google_compute_firewall" "temp_allow_8443_opa_east" {
+  name      = "fw-${var.environment_code}-shared-base-temp-allow-8443-east"
+  project   = local.base_project_id
+  network   = module.base_shared_vpc.network_self_link
+  priority  = 1000
+  direction = "INGRESS"
+
+  dynamic "log_config" {
+    for_each = var.firewall_enable_logging == true ? [{
+      metadata = "INCLUDE_ALL_METADATA"
+    }] : []
+
+    content {
+      metadata = log_config.value.metadata
+    }
+  }
+
+  source_ranges = ["172.16.2.0/28"]
+  target_tags   = ["gke-gke-boa-us-east1-001-1d5c9eff-node"]
```

Comment:
This seems like an autogenerated tags `gke-gke-boa-us-east1-001-1d5c9eff-node` here and throughout?

Reply:
@bharathkkb Probably.. it looks like the tags are following the naming conventions of the VMs, but those are a bit long as well.  Either way, network tags should probably be created up front and the names should be shortened, right?

Reply:
the tags for the clusters are boa-gke1-cluster, boa-gke2-cluster, boa-mci-cluster

Reply:
@rutalreja-deloitte  Yup, I think these are separate tags needed for the nodes.  The cluster tags are all used further down in separate firewall rules.